avro4s-macros/src/main/scala/com/sksamuel/avro4s/SchemaFor.scala

```diff
@@ -370,28 +367,21 @@ object SchemaFor {
   }
 
   private def fieldBuilder(name: String, annos: Seq[Anno], schema: Schema, default: Any): Schema.Field = {
-    def toNode(value: Any): JsonNode = value match {
-      case x: Int => new IntNode(x)
-      case x: Long => new LongNode(x)
-      case x: Boolean => BooleanNode.valueOf(x)
-      case x: Double => new DoubleNode(x)
-      case x: Seq[_] =>
-        val arrayNode = new ArrayNode(JsonNodeFactory.instance)
-        x.foreach(z => arrayNode.add(toNode(z)))
-        arrayNode
-      case x: Map[String, _] =>
-        val objectNode = new ObjectNode(JsonNodeFactory.instance)
-        x.foreach { case (k, v) =>
-          objectNode.put(k, toNode(v))
-        }
-        objectNode
-      case Some(x) => toNode(x)
-      case None => NullNode.instance
-      case _ => new TextNode(value.toString)
+    def toDefaultValue(value: Any): Any = value match {
+      case x: Int => x
+      case x: Long => x
+      case x: Boolean => x
+      case x: Double => x
+      case x: Seq[_] => x.asJava
+      case x: Map[_, _] => x.asJava
+      case Some(x) => x
+      case None => JsonProperties.NULL_VALUE
+      case _ => value.toString
```

Comment:
This is the riskiest change in the PR. The `Schema.Field` constructors that took a `JsonNode` argument for the default value have all been deprecated. There was a unit test in place that covered the expected behavior here, and it still passes with this change.